packages/flutter/lib/src/material/chip.dart

```diff
@@ -150,16 +154,22 @@ class Chip extends StatelessWidget {
 
     return new Semantics(
       container: true,
-      child: new Container(
-        height: _kChipHeight,
-        padding: new EdgeInsetsDirectional.only(start: startPadding, end: endPadding),
-        decoration: new BoxDecoration(
-          color: backgroundColor ?? Colors.grey.shade300,
-          borderRadius: new BorderRadius.circular(16.0),
-        ),
-        child: new Row(
-          children: children,
-          mainAxisSize: MainAxisSize.min,
+      child: new ConstrainedBox(
+        constraints: const BoxConstraints(minHeight: _kChipHeight),
+        child: new Container(
+          padding: new EdgeInsetsDirectional.only(start: startPadding, end: endPadding),
+          decoration: new BoxDecoration(
+            color: backgroundColor ?? Colors.grey.shade300,
+            shape: BoxShape.pill,
+          ),
```

Comment:
we should use the new decoration infrastructure, rather than adding a new enum value